app/Models/Device.php

```diff
@@ -252,4 +256,46 @@ public function setIpAttribute($ip)
         $this->attributes['ip'] = inet_pton($ip);
     }
 
+    // ---- Query scopes ----
+
+    public function scopeDeviceUp($query)
```

Comment:
Would this be better with out the device part or do you like better with?  scopeIsUp?